daemon/qrexec-daemon.c

```diff
@@ -1139,15 +1159,24 @@ _Noreturn static void handle_execute_service_child(
                            5 /* 5 second timeout */,
                            false /* return 0 not remote status code */));
     } else {
-        char *buf;
-        if (strncmp("@dispvm:", target, sizeof("@dispvm:") - 1) == 0) {
-            disposable = true;
-            buf = qubesd_call(target + 8, "admin.vm.CreateDisposable", "", &resp_len);
+        bool const use_uuid = target_uuid != NULL;
+        const char *const selected_target = use_uuid ? target_uuid : target;
+        bool const disposable = strncmp(QREXEC_DISPVM_PREFIX, selected_target, QREXEC_DISPVM_PREFIX_SIZE) == 0;
+        if (disposable) {
+            char *buf = qubesd_call2(selected_target + QREXEC_DISPVM_PREFIX_SIZE,
+                                     "admin.vm.CreateDisposable", "",
+                                     "uuid", use_uuid ? 4 : 0, &resp_len);
             if (!buf) // error already printed by qubesd_call
                 daemon__exit(126);
             if (memcmp(buf, "0", 2) == 0) {
-                /* we exec later so memory leaks do not matter */
-                target = buf + 2;
```

Comment:
After last force-push the target isn't updated with name/uuid of the just started DispVM anymore.